src/python_testing/TC_CADMIN_1_3_4.py

```diff
@@ -0,0 +1,365 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import asyncio
+import random
+from time import sleep
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.exceptions import ChipStackError
+from chip.interaction_model import Status
+from chip.tlv import TLVReader
+from matter_testing_infrastructure.chip.testing.matter_testing import (MatterBaseTest, TestStep, async_test_body,
+                                                                       default_matter_test_main)
+from mdns_discovery import mdns_discovery
+from mobly import asserts
+
+opcreds = Clusters.OperationalCredentials
+nonce = random.randbytes(32)
+
+
+class TC_CADMIN_1_3_4(MatterBaseTest):
+    async def get_fabrics(self, th: ChipDeviceCtrl) -> int:
+        OC_cluster = Clusters.OperationalCredentials
+        if th == self.th2:
+            th2_fabric_info = await th.ReadAttribute(nodeid=self.dut_node_id, fabricFiltered=True, attributes=[(0, OC_cluster.Attributes.Fabrics)])
+            th2_fabric_data = list(th2_fabric_info.values())[0]
+            th2_fabric_data = list(th2_fabric_data.values())[0]
+            fabric_info = vars(list(th2_fabric_data.values())[1][0])
+
+        else:
+            fabric_info = await self.read_single_attribute_check_success(dev_ctrl=th, fabric_filtered=True, cluster=OC_cluster, attribute=OC_cluster.Attributes.Fabrics)
+        return fabric_info
+
+    async def get_rcac_decoded(self, th: str) -> int:
```

Comment:
is sending the csr request here just so you can issue a noc chain so you can get the rcac? If so, you shouldn't need to jump through so many hoops - the CA manager is

self.certificate_authority_manager and it maintains a list of the active roots: activeCaList

Reply:
I have resolved the above comment, I was able to use the self.certificate_authority_manager.activeCaList method as you mentioned above in the test module and was able to remove the function that was being used previously.

One caveat, however, since running the test 2 iterations in CI without factory-reset inbetween. I have added a try and except clause to TC_CADMIN_1_4 test to make sure that it uses the correct root public key from self.certificate_authority_manager.activeCaList during the second iteration.